pkg/handler/metricsconfigurator/handler.go

```diff
@@ -261,6 +270,26 @@ func (h Handler) getDatasource(token *string, observeDomainName string, client *
 	return bodyBytes, nil
 }
 
+func (h Handler) makeMetricGroups(MetricsFilters []types.MetricStreamFilter) [][]types.MetricStreamFilter {
+	// create appropriate number of metric streams for metrics, assign metrics to streams
+
+	currentNameCount := 0
+	filterGroups := make([][]types.MetricStreamFilter, 0)
+	currentFilterGroup := make([]types.MetricStreamFilter, 0)
+	for _, filter := range MetricsFilters {
```

Comment:
feels like we could do a flatten and then a chunk here no? https://pkg.go.dev/slices@master#Chunk

Reply:
ah, I didn't know this existed, but it's much cleaner, I'll refactor accordingly!

Reply:
Ah wait, I think that Chunk would not work here - two reasons:
1. We would end up with metrics from the same namespace in different streams. We'd need to make a `types.MetricStreamFilter` for the same namespace in two different streams.
2. AWS counts the namespace itself towards the 1000 (see [here](https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_MetricStreamFilter.html)) so there would be some pretty gnarly cases (adding the namespaces in takes us over the boundary).

Reply:
yeah so then this basically boils down to binpacking, theoretically its possible to be more efficient but we don't really care about how many metric streams it takes us so this simple approach is probably best